src/Symfony/Bridge/Doctrine/Form/Type/DoctrineType.php

```diff
@@ -132,6 +132,13 @@ public function configureOptions(OptionsResolver $resolver)
                 $hash = null;
                 $qbParts = null;
 
+                if (null !== $options['query_builder']) {
+                    $entityLoader = $this->getLoader($options['em'], $options['query_builder'], $options['class']);
+                } else {
+                    $queryBuilder = $options['em']->getRepository($options['class'])->createQueryBuilder('e');
+                    $entityLoader = $this->getLoader($options['em'], $queryBuilder, $options['class']);
```

Comment:
this logic should not be here. It should stay after the cache handling, to do it only when necessary.